src/ServiceControl/MessageRedirects/Handlers/RemoveMessageRedirectHandler.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace ServiceControl.MessageRedirects.Handlers
+{
+    using NServiceBus;
+    using Raven.Client;
+    using ServiceControl.Contracts.MessageRedirects;
+    using ServiceControl.MessageRedirects.InternalMessages;
+
+    public class RemoveMessageRedirectHandler : IHandleMessages<RemoveMessageRedirect>
+    {
+        public IDocumentSession Session { get; set; }
+
+        public IBus Bus { get; set; }
+
+        public void Handle(RemoveMessageRedirect message)
+        {
+            var messageRedirect = Session.Load<MessageRedirect>(MessageRedirect.GetDocumentIdFromMessageRedirectId(message.MessageRedirectId));
```

Comment:
You can avoid the `Load` by using:
`Store.DatabaseCommands.Delete((MessageRedirect.GetDocumentIdFromMessageRedirectId(message.MessageRedirectId), typeof(MessageRedirect), false), null)`.
See DeleteCustomCheckHandler class